consensus/safety-rules/src/persistent_safety_storage.rs

```diff
@@ -117,7 +143,7 @@ impl PersistentSafetyStorage {
         Ok(())
     }
 
-    pub fn last_voted_round(&self) -> Result<Round> {
+    pub fn last_voted_round(&mut self) -> Result<Round> {
```

Comment:
why does this become mutable?

Reply:
I would prefer this retains its non-mutable interface by either using locks or refcell in the cache.

Reply:
It seems that we can't use a RefCell since KVStorage requires the `Sync` trait... I think locks might be possible